README.txt
ejabberd-modules is a collaborative development area for ejabberd modules developers and users.

For users
=========

You need to have Erlang installed.

To use an ejabberd module coming from this repository:
 - Read the module specific README to see if special steps are needed to deploy it.
 - Run "./build.sh" or "build.bat" in the root (usually trunk directory) of the wanted module.
 - Copy generated .beam files from the ebin directory to the directory where your ejabberd .beam files are.
 - Use the configuration file examples provided in the conf dir to update your ejabberd.cfg configuration file.

For developers
==============

The following organisation has been set-up for the development:

- Each module has its own SVN structure (trunk/branches/tags) to allow independant versioning.

- Development and compilation of module should be possible without ejabberd SVN, as long as developers check-out the ejabberd-dev module. This module contains include file to make compilation possible.

- The module directory structure is usually the following:
 README.txt: Module description
 LICENSE.txt: License for the module
 Emakefile: Erlang makefile to build the module (prefered way, if no dependancies on C code, as build will thus works on Windows)
 doc/: Documentation dir
 src/: Source directory
 ebin/: empty (Target directory for the build).
 conf/: Directory containing example configuration for your module.
 build.sh: *nix build script.
 build.bat: Windows build script.

- Module developers should put in the README if the module has known incompatibilities with other modules (for example, by modifying the same main ejabberd modules).


jorge/Changelog
------------------------------------------------------------------------
r116 | kolargol | 2007-07-13 19:40:38 +0200 (Fri, 13 Jul 2007) | 3 lines

removed unused functions


------------------------------------------------------------------------
r115 | kolargol | 2007-07-13 19:36:29 +0200 (Fri, 13 Jul 2007) | 2 lines

added patch for mod_logdb that provides support for fulltext index

------------------------------------------------------------------------
r114 | kolargol | 2007-07-13 18:28:02 +0200 (Fri, 13 Jul 2007) | 2 lines

updated READEME file

------------------------------------------------------------------------
r113 | kolargol | 2007-07-13 18:25:24 +0200 (Fri, 13 Jul 2007) | 2 lines

updated help file

------------------------------------------------------------------------
r112 | kolargol | 2007-07-13 17:46:51 +0200 (Fri, 13 Jul 2007) | 4 lines

- prevent of displaying score row if not nessesery
- minimal word lenght is now 3 characters (before 4)


------------------------------------------------------------------------
r111 | kolargol | 2007-07-13 17:35:14 +0200 (Fri, 13 Jul 2007) | 4 lines

finished search engine v2. fixed a lot of bugs in search engine. Now i use compleatly different search philosophy ;)
code base need some cleanups..


------------------------------------------------------------------------
r110 | kolargol | 2007-07-12 15:37:50 +0200 (Thu, 12 Jul 2007) | 3 lines

quick fix to old search engine (fixed bug where spliting lines was not displayed)


------------------------------------------------------------------------
r109 | kolargol | 2007-07-12 12:39:00 +0200 (Thu, 12 Jul 2007) | 5 lines

added search engine version 2 - this initial commit before enabling it for users. I can say "search refined" version. It uses temp tables for analyzing
score of search results as well as limit displaying it to 200 entries. Search engine no longer shows all results - it shows only relevant ones.
It is powered by full-text index working in BOOLEAN mode so you can use advanced queries to ass score for particular words.


------------------------------------------------------------------------
r108 | kolargol | 2007-07-11 16:20:53 +0200 (Wed, 11 Jul 2007) | 5 lines

- new search engine is here - need testing - i have some mixed feelings as mod_logdb design not allow sorting results by score. I haven't found way to do that - yet :(
- fixed bug with wrong order in search engine
- temporary disabled text highlighting 


------------------------------------------------------------------------
r107 | kolargol | 2007-07-10 15:23:07 +0200 (Tue, 10 Jul 2007) | 4 lines

a lot of internal changes - all source files have been modified. I hope there is not lot of errors and bugs...This commit includes several minr fixes and improvements
and is almost ready for beta release. Also i have added compatibile mod_logdb source file as preparation for public access.


------------------------------------------------------------------------
r106 | kolargol | 2007-07-10 13:21:30 +0200 (Tue, 10 Jul 2007) | 5 lines

- fixed ordering bug in browser
- make all parameters globaly configurable via config.php
- lot of code changes and cleanups


------------------------------------------------------------------------
r105 | kolargol | 2007-07-10 09:36:58 +0200 (Tue, 10 Jul 2007) | 3 lines

global configuration changes


------------------------------------------------------------------------
r104 | kolargol | 2007-07-09 20:19:58 +0200 (Mon, 09 Jul 2007) | 3 lines

adopted main settings from config.php


------------------------------------------------------------------------
r103 | kolargol | 2007-07-09 19:47:47 +0200 (Mon, 09 Jul 2007) | 2 lines

added preliminary config file

------------------------------------------------------------------------
r102 | kolargol | 2007-07-09 15:57:57 +0200 (Mon, 09 Jul 2007) | 6 lines

- added tooltips info search results - now when you hover over lines you see full JID of the person...
- fixed some internal debug info
- fixed some checks in search functions - i see now that it should be completly rewritten...
Known issue: I have observed some "Internal Errors", i started to debug that....


------------------------------------------------------------------------
r101 | kolargol | 2007-07-05 12:31:08 +0200 (Thu, 05 Jul 2007) | 3 lines

added database size info - visible only for admins


------------------------------------------------------------------------
r100 | kolargol | 2007-07-05 10:47:56 +0200 (Thu, 05 Jul 2007) | 3 lines

internal checks and improvements...


------------------------------------------------------------------------
r99 | kolargol | 2007-07-05 10:16:08 +0200 (Thu, 05 Jul 2007) | 3 lines

updated readme


------------------------------------------------------------------------
r98 | kolargol | 2007-07-05 09:55:14 +0200 (Thu, 05 Jul 2007) | 2 lines

removed ugly highlighting php function and replace it by javascript - i think more safe higlighting code

------------------------------------------------------------------------
r97 | kolargol | 2007-07-04 20:09:06 +0200 (Wed, 04 Jul 2007) | 2 lines

added "tracking" option to tooltips

------------------------------------------------------------------------
r96 | kolargol | 2007-07-04 20:04:00 +0200 (Wed, 04 Jul 2007) | 2 lines

fixed order (thx Jabluszko)

------------------------------------------------------------------------
r95 | kolargol | 2007-07-04 19:42:13 +0200 (Wed, 04 Jul 2007) | 2 lines

updated TODO

------------------------------------------------------------------------
r94 | kolargol | 2007-07-04 19:37:56 +0200 (Wed, 04 Jul 2007) | 3 lines

- added back - stats per day
- little lang redone

------------------------------------------------------------------------
r93 | kolargol | 2007-07-04 19:31:40 +0200 (Wed, 04 Jul 2007) | 2 lines

added forgoten legal note from DynamicDrive

------------------------------------------------------------------------

jorge/admin.php
<?
/*
Jorge - frontend for mod_logdb - ejabberd server-side message archive module.

Copyright (C) 2007 Zbigniew Zolkiewski

This program is free software; you can redistribute it and/or
modify it under the terms of the GNU General Public License
as published by the Free Software Foundation; either version 2
of the License, or (at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program; if not, write to the Free Software
Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.

*/
require ("headers.php");
include ("upper.php");

if ($admin_name!=$token) { print 'no access'; exit; }

$today="";

$today = date("Y-n-d");

$top_ten_talkers_today="select at, owner_id, count from `messages-stats_$xmpp_host` where at = '$today' order by count desc limit 10";
$top_ten_talkers_yesterday="select at, owner_id, count from `messages-stats_$xmpp_host` where at = (select date_format((date_sub(curdate(),interval 1 day)), \"%Y-%c-%d\")) order by count desc limit 10";

$month_stats="select count(owner_id) users_total, at, sum(count) as messages from `messages-stats_$xmpp_host` group by at order by str_to_date(at,'%Y-%m-%d') desc limit 30";
$result=mysql_query($month_stats);
while ($entry=mysql_fetch_array($result)) {
	$i++;
	$d[$i] = $entry[messages];
	$e[$i] = $entry[users_total];
	
}

$maximum_a = max($e);
$maximum_b = max($d);

print '<table class="ff">'."\n";
print '<tr><td>'."\n";
print '<p><b>Number of user using message archiving:</b></p>'."\n";
print '<div id="chart" class="chart" style="width: 1000px; height: 200px;"></div>'."\n";
print '<p><b>Number of messages logged by server:</b></p>';
print '<div id="messages" class="chart" style="width: 1000px; height: 200px;"></div>'."\n";
print '</td>';
print '<td style="padding-left: 30px">'."\n";
print '<div><b>Top ten talkers today:</b><br><br>'."\n";
$result=mysql_query($top_ten_talkers_today);
$i=0;
while ($entry=mysql_fetch_array($result)) {
	
	$i++;
	print "<b>".$i.".</b> ".get_user_name($entry[owner_id],$xmpp_host)."@".$xmpp_host." (<i>$entry[count]</i>)<br>"."\n";

}
print '</div>'."\n";
print '<br><hr size="1" noshade="" color="#cccccc"/><br>'."\n";
$i=0;
print '<div><b>Top ten talkers yesterday:</b><br><br>'."\n";
$result=mysql_query($top_ten_talkers_yesterday);
while ($entry=mysql_fetch_array($result)) {

	$i++;
	print "<b>".$i.".</b> ".get_user_name($entry[owner_id],$xmpp_host)."@".$xmpp_host." (<i>$entry[count]</i>)<br>"."\n";
	
}

print '</td>'."\n";
print '</tr></table>'."\n";

?>

<script type="text/javascript">
    	function draw() {
    		var b = new Chart(document.getElementById('chart'));
				b.setDefaultType(CHART_LINE);
				b.setGridDensity(32, 10);
				<? print "b.setVerticalRange(0, $maximum_a);"; ?>	
				b.setHorizontalLabels(['30','29','28','27','26','25','24','23','22','21','22','21','20','19','18','17','16','15','14','13','12','11','10','9','8','7','6', '5', '4', '3', '2', '1']);
				<? print "b.add ('Last 30 days', '#ff0000', ["; 
			
				for ($z=30; $z>0; $z--) {
						
						print "$e[$z],$e[$z],";
					
					}
				print "]);\n";


?>
				b.draw();
    	
	
	var b = new Chart(document.getElementById('messages'));
				b.setDefaultType(CHART_LINE);
				b.setGridDensity(32, 10);
				<? print "b.setVerticalRange(0, $maximum_a);"; ?>	
				b.setHorizontalLabels(['30','29','28','27','26','25','24','23','22','21','22','21','20','19','18','17','16','15','14','13','12','11','10','9','8','7','6', '5', '4', '3', '2', '1']);
				<? print "b.add ('Last 30 days', '#3480ff', ["; 
			
				for ($z=30; $z>0; $z--) {
						
						print "$d[$z],$d[$z],";
					
					}
				print "]);\n";


?>
				b.draw();
    	

	
	}
    	
    	window.onload = function() {
    		draw();
    	};

</script>



<?

include ("footer.php");
?>
